From 41bb0b5a6d974ca50e0c3fb032a60514373c4f17 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 22 Nov 2016 11:00:02 +0200
Subject: [PATCH 1118/2241] fix: udc: fix null pointer access when UDC suspend

fix null pointer access in UDC driver while running suspend to RAM

Change-Id: I15b74bf530f9630b4c4e9cbb4bc62e626aee90f8
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33951
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/usb/gadget/udc/mv_udc_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/udc/mv_udc_core.c b/drivers/usb/gadget/udc/mv_udc_core.c
index 44b25fb..faac1a8 100644
--- a/drivers/usb/gadget/udc/mv_udc_core.c
+++ b/drivers/usb/gadget/udc/mv_udc_core.c
@@ -2432,7 +2432,7 @@ static int mv_udc_suspend(struct device *dev)
 	if (udc->transceiver)
 		return 0;
 
-	if (udc->pdata->vbus && udc->pdata->vbus->poll)
+	if (udc->pdata && udc->pdata->vbus && udc->pdata->vbus->poll)
 		if (udc->pdata->vbus->poll() == VBUS_HIGH) {
 			dev_info(&udc->dev->dev, "USB cable is connected!\n");
 			return -EAGAIN;
-- 
2.7.4

